### PR TITLE
Add better assert messages for f32/f64 clamps

### PR DESCRIPTION
**Pull request by Xaeroxe**

### CODE REVIEW

File: library/core/src/num/f64.rs [hunk 1]
```diff
@@ -1389,7 +1389,10 @@ impl f64 {
     #[stable(feature = "clamp", since = "1.50.0")]
     #[inline]
     pub fn clamp(mut self, min: f64, max: f64) -> f64 {
-        assert!(min <= max);
+        assert!(
+            min <= max,
+            "min > max, or either was NaN. This is not allowed. min = {min:?}, max = {max:?}"
+        );
```
Comment (joshtriplett) [inline library/core/src/num/f64.rs:1395]: These messages may be seen by the end user, but they're typically a programming error, so let's avoid phrasing that might be misinterpreted. An assertion failure already makes it clear something wasn't allowed.
```suggestion
        assert!(
            min <= max,
            "min > max, or either was NaN. min = {min:?}, max = {max:?}"
        );
```

**Comment by joshtriplett** [comment]
Minor tweak to the message, but otherwise LGTM. r=me with the suggestions (or similar) applied.

File: library/core/src/num/f32.rs [hunk 2]
```diff
@@ -1391,7 +1391,10 @@ impl f32 {
     #[stable(feature = "clamp", since = "1.50.0")]
     #[inline]
     pub fn clamp(mut self, min: f32, max: f32) -> f32 {
-        assert!(min <= max);
+        assert!(
+            min <= max,
+            "min > max, or either was NaN. This is not allowed. min = {min:?}, max = {max:?}"
```
[the remark this answers is not in the document]
Comment (Xaeroxe) [reply library/core/src/num/f32.rs]: Done

File: library/core/src/num/f64.rs (hunk 1 above)
Comment (Xaeroxe) [reply library/core/src/num/f64.rs]: Done

**Comment by Xaeroxe** [comment]
I'd like this to go into the 1.66 release. With the inclusion of the new clippy lint `manual_clamp` in 1.66, more people are going to trigger this panic message than they would in 1.65

**Comment by Xaeroxe** [comment]
@joshtriplett this is ready for re-review

**Comment by JohnCSimon** [comment]
ping from triage:
@Xaeroxe 

FYI: when a PR is ready for review, send a message containing 
`@rustbot ready` to switch to `S-waiting-on-review` so the PR is in the reviewer's backlog.

doing this for you - 
@rustbot ready

**Comment by Xaeroxe** [comment]
Thank you!

**Comment by ChrisDenton** [comment]
lgtm, thanks!

@bors r+

**Pull request merged**